api/apiclient.go

```diff
@@ -891,7 +938,15 @@ func dialWebsocketMulti(ctx context.Context, addrs []string, path string, opts d
 			continue
 		}
 		tried[ipStr] = true
-		err = startDialWebsocket(ctx, try, ipStr, resolvedAddr.host, path, opts)
+
+		// Make a copy of the controller's root URL before we append the API path.
+		// The original URL should be preserved because it too may contain a path segment
+		// if a controller is hosted on a subpath, e.g. https://controller.com/foo/bar.
+		urlWithAPIPath := &url.URL{}
+		*urlWithAPIPath = *resolvedAddr.url
+		urlWithAPIPath.Path = gopath.Join(urlWithAPIPath.Path, apiPath)
+
+		err = startDialWebsocket(ctx, try, resolvedAddr.url, urlWithAPIPath, ipStr, opts)
```

Comment:
Is resolvedAddr.url *guaranteed* to not be nil? 

This can be simplified

```
urlWithAPIPath = *resolvedAddr.url
urlWithAPIPath.Path = gopath.Join(urlWithAPIPath.Path, apiPath)

err = startDialWebsocket(ctx, try, resolvedAddr.url, &urlWithAPIPath, ipStr, opts)
```